Adds Swagger support for PAB API

kelizarov opened the pull request:
<!--
Here are some checklists you may like to use. Use your judgement.

This is just a checklist, all the normative suggestions are covered in more detail in CONTRIBUTING.
-->
Pre-submit checklist:
- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

goosedb committed:
swagger

goosedb committed:
orphans

kelizarov commented on plutus-ledger-api/src/Plutus/V1/Ledger/Scripts.hs:
```diff
@@ -85,10 +85,16 @@ import           PlutusTx.Prelude
   deriving stock Generic
 
+instance Swagger.ToSchema Script where
+    declareNamedSchema _ = Haskell.pure $ Swagger.NamedSchema (Just "Plutus Script") Haskell.mempty
```
Здесь `Plutus Script` лучше сделать `PlutusScript`. Сваггер ругается что имя имеет пробел.

kelizarov commented on plutus-ledger-api/src/Plutus/V1/Ledger/Scripts.hs:
```diff
@@ -237,6 +243,11 @@ newtype Datum = Datum { getDatum :: BuiltinData  }
+    declareNamedSchema _ = Haskell.pure $ Swagger.NamedSchema (Just "Datum") Haskell.mempty
+
+
+
```
Надо бы пофиксить форматтинг по всем файлам.

goosedb replied:
оно само при коммите форматирует

goosedb committed:
swagger

goosedb committed:
swagger

goosedb committed:
Merge remote-tracking branch 'metalamp/master' into swagger

goosedb committed:
Merge remote-tracking branch 'metalamp/master' into swagger

kelizarov commented:
@goosedb вроде норм. А ты не проверял возможность чтобы можно было не-техническую информацию добавить к эндпойнтам?

kelizarov commented:
И сможешь сделать ребейз в ветки чтобы одним коммитом это было и добавить сообщение которое объясняет изменение.

goosedb committed:
migrate swagger2 to openapi3 + ui page

goosedb committed:
migrate swagger2 to openapi3 + ui page

kelizarov commented on plutus-pab/src/Plutus/PAB/Webserver/Server.hs:
```diff
@@ -56,16 +57,19 @@ asHandler PABRunner{runPABAction} = Servant.Handler . ExceptT . fmap (first mapE
-      :<|> WSAPI
+type CombinedAPI t = BasicCombinedAPI t :<|> SwaggerAPI
+
+type BasicCombinedAPI t =
```
Let's call it `BaseCombinedAPI`

kelizarov commented on shell.nix:
```diff
@@ -72,6 +72,7 @@ let
     yq
     z3
     zlib
+    lzma
```
Can you tell here what this is addition is for?

goosedb replied:
servant-swagger-ui requires lzma

kelizarov commented on plutus-pab/src/Plutus/PAB/Webserver/API.hs:
```diff
@@ -30,20 +32,21 @@ type WSAPI =
 -- | PAB client API for contracts of type @t@. An example of @t@ are
 --   * "Builtin" contracts that run in the same process as the PAB (ie. the PAB is compiled & distributed with these contracts)
 type API t walletId -- see note [WalletID type in wallet API]
```
Some API endpoints are missing Description. Maybe we should add description to all of them?

kelizarov commented on orphans/CHANGELOG.md:
```diff
@@ -0,0 +1,5 @@
+# Revision history for orphans
```
We should add a README.md for this package to explain the use-case.

kelizarov commented on orphans/src/Orphans.hs:
```diff
@@ -0,0 +1,18 @@
+{-# LANGUAGE FlexibleInstances  #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE StandaloneDeriving #-}
+module Orphans () where
```
I'm not sure if creating a package just for orphans is enough. Maybe we can call something like `plutus-extra` and put Orphans module there?

I also think that maybe we should add to the naming some information what orphans are for. For example here we can call it `Orpahns.OpenApi` or something.

kelizarov commented on orphans/orphans.cabal:
```diff
@@ -0,0 +1,24 @@
+
+homepage:
+
+license:            NONE
```
Can you please look into how plutus cabal file are strucutred and copy it here?

kelizarov commented on plutus-core/plutus-core/src/PlutusCore/Data.hs:
```diff
@@ -16,22 +16,26 @@ import qualified Codec.CBOR.Magic          as CBOR
-
-The other constructors are various primitives.
--}
+-- | A generic "data" type.
```
I suspect this is auto-formatter doing?

kelizarov commented on plutus-core/plutus-core/src/PlutusCore/Data.hs:
```diff
@@ -16,22 +16,26 @@ import qualified Codec.CBOR.Magic          as CBOR
+import           GHC.Exts                  (IsList (..))
 import           GHC.Generics
 import           Prelude
 
```
There's a blank line between imports. Let's remove it. 

Can you also check the rest of files for this?

kelizarov commented:
There's also a conflict that's need to be resolved :)

goosedb committed:
readme for orphans package

goosedb committed:
move imports + move orphans module

goosedb committed:
Merge branch 'master' into swagger

goosedb committed:
more description for endpoints

goosedb committed:
updateMaterialized nix

goosedb committed:
Merge branch 'master' into swagger

kelizarov committed:
Merge branch 'master' into swagger

kelizarov committed:
Removes OpenAPI import for notes/Ledger.hs.

kelizarov committed:
Removed lzma depedency from shell.nix as it gets resolved by haskell.nix

kelizarov committed:
Removes unused packages in plutus-ledger.

kelizarov committed:
Merge branch 'master' into swagger

kelizarov committed:
Adds missing ToSchema deriviation.

kelizarov committed:
Adds ToSchema instance for MarloweContract

Also adds openapi3 package for marlowe exe and test targets.

kelizarov committed:
Adds orphan instance for OpenAPI in plutus-ledger

This commit removes orphan package that handles some orphan instance for ToSchema typeclass.
Also removes openapi3 from plutus-ledger-api and plutus-contract from dependency tree.

kelizarov committed:
Merge remote-tracking branch 'origin/master' into swagger

kelizarov committed:
Removes unused import from Orphans module

kelizarov committed:
Fixes codestyle issues

kelizarov committed:
Removes unused Generic import from notes

kelizarov committed:
Merge branch 'master' into swagger

kelizarov committed:
Merge branch 'master' into swagger

kelizarov committed:
Merge branch 'master' into swagger

kelizarov committed:
Merge branch 'master' into swagger

kelizarov committed:
Removes unnescerary imports and cleans code.

kelizarov committed:
Removes unused INCOHERENT pragma.
